be/src/olap/rowset/segment_v2/segment_iterator.cpp

```diff
@@ -652,6 +652,10 @@ void SegmentIterator::_vec_init_lazy_materialization() {
     std::set<ColumnId> del_cond_id_set;
     _opts.delete_condition_predicates->get_all_column_ids(del_cond_id_set);
 
+    if (config::enable_low_cardinality_optimize) {
+        _check_dict_columns_can_be_used_in_vec_predicates();
```

Comment:
This method is not related with lazy materialization, should not in _vec_init_lazy_materialization.
Pls add this in 
Status SegmentIterator::_init(bool is_vec) {
    SCOPED_RAW_TIMER(&_opts.stats->block_init_ns);
    DorisMetrics::instance()->segment_read_total->increment(1);
    // get file handle from file descriptor of segment
    fs::BlockManager* block_mgr = fs::fs_util::block_manager(_segment->_path_desc);
    RETURN_IF_ERROR(block_mgr->open_block(_segment->_path_desc, &_rblock));
    _row_bitmap.addRange(0, _segment->num_rows());
    RETURN_IF_ERROR(_init_return_column_iterators());
    RETURN_IF_ERROR(_init_bitmap_index_iterators());
    // z-order can not use prefix index
    if (_segment->_tablet_schema->sort_type() != SortType::ZORDER) {
        RETURN_IF_ERROR(_get_row_ranges_by_keys());
    }
    RETURN_IF_ERROR(_get_row_ranges_by_column_conditions());
    if (is_vec) {
        _vec_init_lazy_materialization();
        _vec_init_char_column_id();
    } else {
        _init_lazy_materialization();
    }
    _range_iter.reset(new BitmapRangeIterator(_row_bitmap));
    return Status::OK();
}